pkg/dogstatsd/replay/state.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package replay
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var (
+	mux    sync.RWMutex
+	pidMap map[int32]string
+
+	errPidMapUnavailable    = errors.New("no pid map has been set for this replay")
+	errContainerUnavailable = errors.New("specified pid is not associated to any container")
+)
+
+// SetPidMap sets the map with the pid - containerID relations
+func SetPidMap(m map[int32]string) {
+	mux.Lock()
+	defer mux.Unlock()
+
+	pidMap = map[int32]string{}
+	for pid, containerID := range m {
+		pidMap[pid] = containerID
+	}
+}
+
+// ContainerIDForPID returns the matching container id for a pid, or an error if not found.
+func ContainerIDForPID(pid int32) (string, error) {
+	mux.RLock()
+	defer mux.RUnlock()
+
+	if pidMap == nil {
+		return "", errPidMapUnavailable
+	}
```

Comment:
We should lock after this

Reply:
Sorry, I'm not sure I follow? Why should we lock? Aren't we already holding an `RLock` there and all we're doing is reading?

Reply:
I meant that we should only acquire the lock after checking the `pidMap` to avoid locking if we're going to return right away.

Reply:
But the `pidMap` will be nil until the first time it's set, so it's part of the race condition. Without the `RLock()` there's a race between `pidMap` being nil and actually being something, hence why I still think we should lock before checking if it's nil.

Reply:
my bad ... I missread the code I thought you were checking the parameter....